coderd/httpmw/cors_test.go

```diff
@@ -0,0 +1,87 @@
+package httpmw_test
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/coder/coder/coderd/httpapi"
+	"github.com/coder/coder/coderd/httpmw"
+)
+
+func TestWorkspaceAppCors(t *testing.T) {
```

Comment:
How does this look for tests?  We could make actual requests but since the PR is just adding headers if the host/origin match up I think just testing the middleware gets us there.

I am happy to go down that route if we want though.  Looks like I would use the `apptest` package?

Reply:
Well one other thing it does, if there is an options request it returns early with a 200.

Reply:
I should test that options behavior too, one second.

Reply:
OK good to go.